Spigot-API-Patches/0314-Add-LivingEntity-hasLineOfSight-Location.patch

```diff
@@ -0,0 +1,30 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: TwoLeggedCat <80929284+TwoLeggedCat@users.noreply.github.com>
+Date: Sat, 29 May 2021 14:33:18 -0500
+Subject: [PATCH] Add LivingEntity#hasLineOfSight(Location)
+
+
+diff --git a/src/main/java/org/bukkit/entity/LivingEntity.java b/src/main/java/org/bukkit/entity/LivingEntity.java
+index 9f0645dc5f76ee9ef73d88f768025429e5a9edf7..257874f61756f1576d04fbd2057234e48229e817 100644
+--- a/src/main/java/org/bukkit/entity/LivingEntity.java
++++ b/src/main/java/org/bukkit/entity/LivingEntity.java
+@@ -483,6 +483,19 @@ public interface LivingEntity extends Attributable, Damageable, ProjectileSource
+      */
+     public boolean hasLineOfSight(@NotNull Entity other);
+ 
++    // Paper start
++    /**
++     * Checks whether the living entity has block line of sight to the given block.
++     * <p>
++     * This uses the same algorithm that hostile mobs use to find the closest
++     * player.
```

Comment:
Then there must be a comment for conflict on change on said algorithm.

Reply:
Sorry, I'm not entirely sure what you mean by that. Could you elaborate?

Reply:
Add a comment right above the code you copied from (like "Used in ... method"), so that if the original code changes, the patch fill fail to apply due to a conflict.

Reply:
Like so?

Reply:
Almost :p Better to put it above the `return this.world.rayTrace(...`, since right now it isn't shown in the patch and wouldn't cause a conflict on change

Reply:
So... like so? 😄